dask/optimization.py

```diff
@@ -25,8 +25,8 @@ def cull(dsk, keys):
     Examples
     --------
 
-    >>> d = {'x': 1, 'y': (inc, 'x'), 'out': (add, 'x', 10)}    # doctest: +SKIP
-    >>> dsk, dependencies = cull(d, 'out')                      # doctest: +SKIP
+    >>> d = {'x': 1, 'y': (inc, 'x'), 'out': (add, 'x', 10)}
+    >>> dsk, dependencies = cull(d, 'out')
```

Comment:
I think the following lines can be unskipped with the following:

```python
    >>> dsk  # doctest: +ELLIPSIS
    {'out': (<function add at ...>, 'x', 10), 'x': 1}
    >>> dependencies
    {'out': ['x'], 'x': []}
```

Reply:
Great suggestion on using `+ELLIPSIS`! I'll see if I can incorporate that for the other PRs as well.

In addition to that, I see you've flipped the order of the keys in the dictionary, presumably to get the output matching. There are many instances across all 6 PRs where I wasn't able to un-skip a doctest simply because the key ordering is different even though the dict itself is identical. 

Should I simply reorder the dict in the docstring to match the test output? How can I be sure that the specific ordering I get on my machine is the "canonical" one that's going to remain unchanged going forward?

Reply:
> Should I simply reorder the dict in the docstring to match the test output? How can I be sure that the specific ordering I get on my machine is the "canonical" one that's going to remain unchanged going forward?

Yeah good point. I just looked at the code and I think the order should always be key (`out`) then dependencies (`x`) so this one seems fine, but there should probably be a unit test that enforces that order as well.

Reply:
Good to know that for this one instance it's safe to flip the order in the dict. Any suggestions for how to handle the same situation in the other PRs and the (I want to say, close to 100) places where we aren't yet sure whether the key ordering in the output dict is deterministic? Is there a simpler way than looking at the source code for all the relevant functions?

Reply:
I guess you could run the tests a few times and see if they fail. But probably best to just leave potentially unordered dicts skipped for now. I opened an issue to get a sense of whether dask should care about dict ordering https://github.com/dask/dask/issues/7911

Reply:
Did you mean to unskip the lines below as suggested?

Reply:
Ahh I was too caught up in the broader discussion and forgot about the actual code.

Fixed. Thanks for catching.